opsimulator/store.go

```diff
@@ -0,0 +1,63 @@
+package opsimulator
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type L1DepositStore struct {
+	entryByHash map[common.Hash]*types.DepositTx
```

Comment:
some high level thoughts

1. entryByHash should be indexed by something that globally & uniquely identifies a depositTX. the l1 txhash is insufficient because 
2. you should prob maybe store more than just the deposit tx here since the indexer likely wants more info than that (ie. the log where this was emitted for instance)

Reply:
update:
1. using the source hash as the key value, since it uniquely identifies the source of the deposit

Reply:
2. updated the message to contain both the depositTxn and the initiating log in the form of a struct